datadog_checks_dev/datadog_checks/dev/tooling/config_validator/config_block.py

```diff
@@ -0,0 +1,431 @@
+# (C) Datadog, Inc. 2019
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+import re
+
+from .utils import get_indent, is_at_least_indented, is_blank, is_exactly_indented
+from .validator_errors import SEVERITY_WARNING, ValidatorError
+
+# The maximum length authorized for comments. It accounts for text after '##' and does not apply to the @param
+# declaration
+MAX_COMMENT_LENGTH = 120
+
+# The list of accepted type as regex. Anything starting with list is allowed.
+ACCEPTED_VAR_TYPE_REGEX = [
+    "^boolean$",
+    "^string$",
+    "^integer$",
+    "^double$",
+    "^float$",
+    "^string$",
+    "^object$",
+    "^list.*$",
+    "^dictionary$",
+]
+
+# Regex used to parse the fields of the '## @param' declaration
+PARAM_REGEX = "^## @param ([a-zA-Z0-9_\\-]+) +- (.+) - (required|optional)( - default: .*)?$"
+
+# Regex to parse a 'key: value' item
+VAR_REGEX = "^(# |)(\\w+): ([a-zA-Z0-9_><]+)$"
+
+# Regex to identify an object (it ends with a colon). No comment allowed after the colon but the whole line
+# can be a comment.
+OBJECT_REGEX = "^(# |)([a-zA-Z0-9_\\-]+): *$"
+
+# Regex to identify a comment. Note that 'INDENT' needs to be updated with the number of spaces expected
+COMMENT_REGEX = "^ {INDENT}##(.*)$"
+
+# Regex to match any comment with no respect to indentation
+INCORRECTLY_INDENTED_COMMENT_REGEX = "^ *##(.*)$"
+
+
+class ParamProperties:
+    """Class to represent a parameter declared using the '@param' annotation
+    """
+
+    def __init__(self, var_name, type_name, required=True, default_value=None):
+        self.var_name = var_name
+        self.type_name = type_name
+        self.required = required
+        self.default_value = default_value
+
+    @classmethod
+    def parse_from_string(cls, idx, config_lines, indent, errors):
+        if not is_exactly_indented(config_lines[idx], indent):
+            errors.append(ValidatorError("Content is not correctly indented", idx))
+            return None
+
+        current_line = config_lines[idx][indent:]
+
+        if not current_line.startswith("## @param"):
+            errors.append(ValidatorError("Expecting @param declaration", idx))
+            return None
+
+        m = re.match(PARAM_REGEX, current_line)
+        if m is None:
+            errors.append(ValidatorError("Invalid @param declaration", idx))
+            return None
+
+        if m.group(3) == "optional":
+            def_val = None
+            if m.group(4):  # If there is a default value
+                def_val = m.group(4)[12:]
+
+            return ParamProperties(m.group(1), m.group(2), False, def_val)
+
+        return ParamProperties(m.group(1), m.group(2), True)
```

Comment:
Sorry I should have specified that it means you can use `cls`here instead of `ParamProperties`.

Reply:
Oh indeed, I see what you meant now. Thanks